src/orange/univ/detail/SenseCalculator.hh

```diff
@@ -28,22 +28,23 @@ namespace detail
  * simple *and* complex intersection. Instances of this class are specific to a
  * volume, and a position. Calling an instance evaluates the sense of a
  * volume's face with respect to the given position. This class is used to
- * lazily calculate sense during evaluation of a logic expression, contrary to
- * CachedLazySenseCalculator, this class does not cache the calculated sense:
- * potentially recomputing the same sense value multiple time. The advantage
- * is that we do not need to access global memory to store the cached sense.
+ * lazily calculate sense during evaluation of a logic expression.
+ * This class does not cache the calculated sense, that is senses are
+ * recomputed every time it is requested. The advantage of not caching senses
+ * is that we eliminate global memory usage. With infix evaluation, and
+ * short-circuiting logic, the cost of recomputing senses should be small.
```

Comment:
With the latest changes this should be "on-the-fly" rather than "lazy", which usually implies caching.